projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -66,35 +69,45 @@ export class IgxOverlayService {
     /**
      * Shows the provided component.
      */
-    show(component: ElementRef | Type<{}>, settings?: OverlaySettings): string {
-        const id: string = (this._componentId++).toString();
-        settings = Object.assign({}, this._defaultSettings, settings);
+    show(component: ElementRef | Type<{}>, settings?: OverlaySettings, id?: string): string {
+        let info: OverlayInfo;
+        if (id !== undefined) {
+            info = this.getOverlayById(id);
+        }
 
-        const info = this.getOverlayInfo(component);
+        if (!info) {
```

Comment:
Does this mean that if I pass a wrong ID with ref to element already in overlay, it'll still make a new one and move it there? Not a biggie, just curious.

Reply:
At this point we do not support moving an element from overlay to overlay to overlay and so on. And yes it will move the element form one overlay to another